### PR TITLE
Add migrator to pin azure-core-cpp 1.12.0

### PR DESCRIPTION
**Pull request by jdblischak**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

---

xref: https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/6003#issuecomment-2177137980
cc: @h-vetinari

### CODE REVIEW

File: recipe/migrations/azure_core_cpp1120.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+migrator_ts: 1718824596
+__migrator:
+  kind: version
+  migration_number: 1
+  bump_number: 1
+  commit_message: Pin azure-core-cpp 1.12.0
+
+azure_core_cpp:
+  - 1.12.0
```
Comment (h-vetinari) [inline recipe/migrations/azure_core_cpp1120.yaml:9]: Let's add all the pins here that you had in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6048, and make sure that they're the most recent version per feedstock.

**Review by h-vetinari — COMMENTED** [review]
You can think of the migrator as a preview of the global pinning. It allows testing the waters and ensuring all relevant packages can actually be built, before actually adding something to the production pinning.

File: recipe/migrations/azure_core_cpp1120.yaml (hunk 1 above)
Comment (jdblischak) [reply recipe/migrations/azure_core_cpp1120.yaml:9]: Done in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6056/commits/6ffa874d6747c6c80c439128ef92e79c6019bddc

**Review by h-vetinari — APPROVED** [review]

**Comment by h-vetinari** [comment]
Thanks you! :)

**Pull request merged**